### PR TITLE
setup pip-tools for local dev

### PR DESCRIPTION
**Pull request by patjouk**
Related PRs/issues mozillafoundation/mofo-devops#742

Heroku will be using regular pip to install dependencies using the `requirements.txt` file.

I need to rewrite a huge chunk of doc before this PR can be merged.

## Checklist

**Documentation:**
- [x] Is my code documented?
- [x] Did I update the READMEs or wagtail documentation?

**TODOs**:
- [x] Update/clean package.json
- [x] Reconfig CI

### CODE REVIEW

**Comment by Pomax** [comment]
Am I reading this correct if I think it means that `requirements.in` is essentially `Pipfile`, and `requirements.txt` now does what `Pipfile.lock` did? (but without cryptohash verification?)

**Comment by patjouk** [comment]
PR ready for review ✨

**Comment by TheoChevalier** [comment]
Some weird things happened after running `inv catchup`:

Capture d’écran 2020-04-17 à 14 05 33

Then it updated dependencies to use http instead of https?
Capture d’écran 2020-04-17 à 14 01 41

Will keep playing around

**Comment by TheoChevalier** [comment]
Nevermind, that might just have been a docker image not gracefully stopped. Killed the image, then it worked.

However, `makemessages` seem broken, even with a fresh env:
Capture d’écran 2020-04-17 à 14 25 38

**Comment by patjouk** [comment]
I forgot to add the python virtualenv to the list of directories makemessages should ignore!

**Comment by patjouk** [comment]
To test: don't forget to delete your `node_modules`.

**Comment by patjouk** [comment]
I went for running `compilemessages` from `network-api`. That way, it's going in all `locale` dirs of the project:

```
processing file django.po in /app/network-api/locale/pl/LC_MESSAGES
processing file django.po in /app/network-api/locale/de/LC_MESSAGES
processing file django.po in /app/network-api/locale/es/LC_MESSAGES
processing file django.po in /app/network-api/locale/pt/LC_MESSAGES
processing file django.po in /app/network-api/locale/fr/LC_MESSAGES
processing file django.po in /app/network-api/locale/nl/LC_MESSAGES
processing file django.po in /app/network-api/networkapi/buyersguide/locale/pl/LC_MESSAGES
processing file django.po in /app/network-api/networkapi/buyersguide/locale/de/LC_MESSAGES
processing file django.po in /app/network-api/networkapi/buyersguide/locale/es/LC_MESSAGES
processing file django.po in /app/network-api/networkapi/buyersguide/locale/pt/LC_MESSAGES
processing file django.po in /app/network-api/networkapi/buyersguide/locale/fr/LC_MESSAGES
processing file django.po in /app/network-api/networkapi/mozfest/locale/pl/LC_MESSAGES
processing file django.po in /app/network-api/networkapi/mozfest/locale/de/LC_MESSAGES
processing file django.po in /app/network-api/networkapi/mozfest/locale/es/LC_MESSAGES
processing file django.po in /app/network-api/networkapi/mozfest/locale/pt/LC_MESSAGES
processing file django.po in /app/network-api/networkapi/mozfest/locale/fr/LC_MESSAGES
processing file django.po in /app/network-api/networkapi/mozfest/locale/nl/LC_MESSAGES
```

**Comment by TheoChevalier** [comment]
> I went for running `compilemessages` from `network-api`. That way, it's going in all `locale` dirs of the project:
> 
> ```
> processing file django.po in /app/network-api/locale/pl/LC_MESSAGES
> processing file django.po in /app/network-api/locale/de/LC_MESSAGES
> processing file django.po in /app/network-api/locale/es/LC_MESSAGES
> processing file django.po in /app/network-api/locale/pt/LC_MESSAGES
> processing file django.po in /app/network-api/locale/fr/LC_MESSAGES
> processing file django.po in /app/network-api/locale/nl/LC_MESSAGES
> processing file django.po in /app/network-api/networkapi/buyersguide/locale/pl/LC_MESSAGES
> processing file django.po in /app/network-api/networkapi/buyersguide/locale/de/LC_MESSAGES
> processing file django.po in /app/network-api/networkapi/buyersguide/locale/es/LC_MESSAGES
> processing file django.po in /app/network-api/networkapi/buyersguide/locale/pt/LC_MESSAGES
> processing file django.po in /app/network-api/networkapi/buyersguide/locale/fr/LC_MESSAGES
> processing file django.po in /app/network-api/networkapi/mozfest/locale/pl/LC_MESSAGES
> processing file django.po in /app/network-api/networkapi/mozfest/locale/de/LC_MESSAGES
> processing file django.po in /app/network-api/networkapi/mozfest/locale/es/LC_MESSAGES
> processing file django.po in /app/network-api/networkapi/mozfest/locale/pt/LC_MESSAGES
> processing file django.po in /app/network-api/networkapi/mozfest/locale/fr/LC_MESSAGES
> processing file django.po in /app/network-api/networkapi/mozfest/locale/nl/LC_MESSAGES
> ```

Ah, right! Running from /app/network-api/ should cover all cases. I keep forgetting there are two "network-api/networkapi" nested folders :)

**Comment by cadecairos** [comment]
I'll help land this.

**Comment by Pomax** [comment]
I can't seem to get this to run: I did a `docker system prune --all` to make sure I wouldn't have any docker remnants, and I deleted the `node_modules` dir, but then running `inv new-env` yields this:

```
Git\mozilla\foundation.mozilla.org>inv new-env
* Setting default environment variables
* Stripping quotes and making sure your DATABASE_URL and ALLOWED_HOSTS are properly setup
* Stopping project's containers and delete volumes if necessary
cd C:\Users\Mike\Documents\Git\mozilla\foundation.mozilla.org && docker-compose down --volumes
Removing network foundationmozillaorg_default
Network foundationmozillaorg_default not found.
Removing volume foundationmozillaorg_postgres_data
* Building Docker images
cd C:\Users\Mike\Documents\Git\mozilla\foundation.mozilla.org && docker-compose build
postgres uses an image, skipping
Building watch-static-files
Step 1/2 : FROM node:12.16.2-stretch-slim
12.16.2-stretch-slim: Pulling from library/node
Digest: sha256:46f2238e95292921b79232f9e4135a8169a4cdd36ab3b811d662e83c1fdc6dc8
Status: Downloaded newer image for node:12.16.2-stretch-slim
 ---> bb86086cce7c
Step 2/2 : WORKDIR /app/
 ---> Running in b63036499285
Removing intermediate container b63036499285
 ---> 2ed4a9b6731c
Successfully built 2ed4a9b6731c
Successfully tagged foundationmozillaorg_watch-static-files:latest
Building backend
Step 1/5 : FROM python:3.7.7-slim
3.7.7-slim: Pulling from library/python
Digest: sha256:8c7800cbfebf5a441786ba633a1122086fd0ac06edb2d77ff31da8f189b0912e
Status: Downloaded newer image for python:3.7.7-slim
 ---> d86f161fda14
Step 2/5 : RUN apt-get update && apt-get install -y       gettext     --no-install-recommends && rm -rf /var/lib/apt/lists/*
 ---> Running in 4f9d03208735
Get:1 http://deb.debian.org/debian buster InRelease [122 kB]
Get:2 http://security.debian.org/debian-security buster/updates InRelease [65.4 kB]
Get:3 http://deb.debian.org/debian buster-updates InRelease [49.3 kB]
Get:4 http://deb.debian.org/debian buster/main amd64 Packages [7907 kB]
Get:5 http://security.debian.org/debian-security buster/updates/main amd64 Packages [187 kB]
Get:6 http://deb.debian.org/debian buster-updates/main amd64 Packages [7380 B]
Fetched 8338 kB in 2s (4989 kB/s)
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
The following additional packages will be installed:
  gettext-base libcroco3 libglib2.0-0 libgomp1 libicu63 libncurses6 libxml2
Suggested packages:
  gettext-doc autopoint libasprintf-dev libgettextpo-dev
Recommended packages:
  curl | wget | lynx libglib2.0-data shared-mime-info xdg-user-dirs libgpm2
The following NEW packages will be installed:
  gettext gettext-base libcroco3 libglib2.0-0 libgomp1 libicu63 libncurses6
  libxml2
0 upgraded, 8 newly installed, 0 to remove and 2 not upgraded.
Need to get 12.0 MB of archives.
After this operation, 46.6 MB of additional disk space will be used.
Get:1 http://security.debian.org/debian-security buster/updates/main amd64 libicu63 amd64 63.1-6+deb10u1 [8300 kB]        
Get:2 http://deb.debian.org/debian buster/main amd64 gettext-base amd64 0.19.8.1-9 [123 kB]
Get:3 http://deb.debian.org/debian buster/main amd64 libglib2.0-0 amd64 2.58.3-2+deb10u2 [1258 kB]
Get:4 http://deb.debian.org/debian buster/main amd64 libxml2 amd64 2.9.4+dfsg1-7+b3 [687 kB]
Get:5 http://deb.debian.org/debian buster/main amd64 libcroco3 amd64 0.6.12-3 [145 kB]
Get:6 http://deb.debian.org/debian buster/main amd64 libgomp1 amd64 8.3.0-6 [75.8 kB]
Get:7 http://deb.debian.org/debian buster/main amd64 libncurses6 amd64 6.1+20181013-2+deb10u2 [102 kB]
Get:8 http://deb.debian.org/debian buster/main amd64 gettext amd64 0.19.8.1-9 [1303 kB]
debconf: delaying package configuration, since apt-utils is not installed
    Fetched 12.0 MB in 1s (10.7 MB/s)
Selecting previously unselected package gettext-base.
(Reading database ... 6828 files and directories currently installed.)
Preparing to unpack .../0-gettext-base_0.19.8.1-9_amd64.deb ...
Unpacking gettext-base (0.19.8.1-9) ...
Selecting previously unselected package libglib2.0-0:amd64.
Preparing to unpack .../1-libglib2.0-0_2.58.3-2+deb10u2_amd64.deb ...
Unpacking libglib2.0-0:amd64 (2.58.3-2+deb10u2) ...
Selecting previously unselected package libicu63:amd64.
Preparing to unpack .../2-libicu63_63.1-6+deb10u1_amd64.deb ...
Unpacking libicu63:amd64 (63.1-6+deb10u1) ...
Selecting previously unselected package libxml2:amd64.
Preparing to unpack .../3-libxml2_2.9.4+dfsg1-7+b3_amd64.deb ...
Unpacking libxml2:amd64 (2.9.4+dfsg1-7+b3) ...
Selecting previously unselected package libcroco3:amd64.
Preparing to unpack .../4-libcroco3_0.6.12-3_amd64.deb ...
Unpacking libcroco3:amd64 (0.6.12-3) ...
Selecting previously unselected package libgomp1:amd64.
Preparing to unpack .../5-libgomp1_8.3.0-6_amd64.deb ...
Unpacking libgomp1:amd64 (8.3.0-6) ...
Selecting previously unselected package libncurses6:amd64.
Preparing to unpack .../6-libncurses6_6.1+20181013-2+deb10u2_amd64.deb ...
Unpacking libncurses6:amd64 (6.1+20181013-2+deb10u2) ...
Selecting previously unselected package gettext.
Preparing to unpack .../7-gettext_0.19.8.1-9_amd64.deb ...
Unpacking gettext (0.19.8.1-9) ...
Setting up libglib2.0-0:amd64 (2.58.3-2+deb10u2) ...
No schema files found: doing nothing.
Setting up gettext-base (0.19.8.1-9) ...
Setting up libgomp1:amd64 (8.3.0-6) ...
Setting up libicu63:amd64 (63.1-6+deb10u1) ...
Setting up libncurses6:amd64 (6.1+20181013-2+deb10u2) ...
Setting up libxml2:amd64 (2.9.4+dfsg1-7+b3) ...
Setting up libcroco3:amd64 (0.6.12-3) ...
Setting up gettext (0.19.8.1-9) ...
Processing triggers for libc-bin (2.28-10) ...
Removing intermediate container 4f9d03208735
 ---> b9179140cc28
Step 3/5 : ENV PYTHONUNBUFFERED 1
 ---> Running in 2a7c4e3d9f5f
Removing intermediate container 2a7c4e3d9f5f
 ---> c0fa4f8cb2e8
Step 4/5 : ENV PYTHONDONTWRITEBYTECODE 1
 ---> Running in 0ec1b4b2aa86
Removing intermediate container 0ec1b4b2aa86
 ---> c80222d5b706
Step 5/5 : WORKDIR /app/
 ---> Running in 583a58716b4f
Removing intermediate container 583a58716b4f
 ---> 86a5720a6b22
Successfully built 86a5720a6b22
Successfully tagged foundationmozillaorg_backend:latest
* Install Node dependencies
cd C:\Users\Mike\Documents\Git\mozilla\foundation.mozilla.org && docker-compose run --rm watch-static-files npm ins
tall
Creating network "foundationmozillaorg_default" with the default driver
Creating volume "foundationmozillaorg_postgres_data" with default driver
npm WARN tar ENOENT: no such file or directory, open '/app/node_modules/.staging/node-gyp-0f156f01/package.json'
npm WARN tar ENOENT: no such file or directory, open '/app/node_modules/.staging/node-gyp-0f156f01/.jshintrc'
npm WARN tar ENOENT: no such file or directory, open '/app/node_modules/.staging/node-gyp-0f156f01/addon.gypi'
npm WARN tar ENOENT: no such file or directory, open '/app/node_modules/.staging/node-gyp-0f156f01/CHANGELOG.md'
npm WARN tar ENOENT: no such file or directory, futime
npm WARN tar ENOENT: no such file or directory, futime
npm WARN tar ENOENT: no such file or directory, futime
npm WARN tar ENOENT: no such file or directory, futime
npm WARN tar ENOENT: no such file or directory, futime
npm WARN tar ENOENT: no such file or directory, futime
npm WARN tar ENOENT: no such file or directory, open '/app/node_modules/.staging/yaml-edc20044/dist/cst/BlockValue.js'
npm WARN tar ENOENT: no such file or directory, futime
npm WARN tar ENOENT: no such file or directory, open '/app/node_modules/.staging/webpack-85fd7e99/lib/util/cleverMerge.js'
npm WARN tar ENOENT: no such file or directory, open '/app/node_modules/.staging/xregexp-d1fd5e33/src/index.js'
npm WARN tar ENOENT: no such file or directory, open '/app/node_modules/.staging/xregexp-d1fd5e33/lib/addons/matchrecursive.js'
npm WARN tar ENOENT: no such file or directory, open '/app/node_modules/.staging/webpack-85fd7e99/lib/dependencies/CommonJsPlugin.js'
npm WARN tar ENOENT: no such file or directory, open '/app/node_modules/.staging/yaml-edc20044/browser/dist/schema/Collection.js'

[...300+ additional errors here...]

npm WARN tar ENOENT: no such file or directory, open '/app/node_modules/.staging/node-sass-231f7636/src/libsass/src/check_nesting.hpp'
npm WARN tar ENOENT: no such file or directory, lstat '/app/node_modules/.staging/stylelint-b158a89c/lib/rules/function-url-quotes'
npm WARN tar ENOENT: no such file or directory, open '/app/node_modules/.staging/node-sass-231f7636/src/libsass/src/color_maps.cpp'
npm WARN tar ENOENT: no such file or directory, lstat '/app/node_modules/.staging/stylelint-b158a89c/lib/rules/function-url-scheme-blacklist'
npm WARN tar ENOENT: no such file or directory, lstat '/app/node_modules/.staging/stylelint-b158a89c/lib/rules/function-url-scheme-whitelist'
npm WARN tar ENOENT: no such file or directory, lstat '/app/node_modules/.staging/stylelint-b158a89c/lib/rules/function-whitelist'
npm WARN tar ENOENT: no such file or directory, lstat '/app/node_modules/.staging/stylelint-b158a89c/lib/rules/function-whitespace-after'
npm WARN tar ENOENT: no such file or directory, lstat '/app/node_modules/.staging/stylelint-b158a89c/lib/rules/indentation'
npm WARN tar ENOENT: no such file or directory, futime
npm WARN tar ENOENT: no such file or directory, open '/app/node_modules/.staging/stylelint-b158a89c/lib/rules/index.js'   
npm WARN tar ENOENT: no such file or directory, open '/app/node_modules/.staging/moment-d30432dd/min/moment-with-locales.min.js'
npm WARN tar ENOENT: no such file or directory, futime
npm ERR! code E409
npm ERR! 409 Conflict - GET https://registry.npmjs.org/@babel/helper-validator-identifier/-/helper-validator-identifier-7.9.0.tgz

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2020-04-21T20_59_36_420Z-debug.log
```

**Comment by cadecairos** [comment]
@Pomax I'm not sure how you triggered that state, I wasn't able to on Linux or Windows. This seems to look like a problem with an HTTP request to npm, maybe it will not happen on a retry?

File: README.md [hunk 1]
```diff
@@ -24,43 +22,46 @@
...
 ### Setup steps
 
 Run the following terminal commands to get started:
 
 - `git clone https://github.com/mozilla/foundation.mozilla.org.git`
 - `cd foundation.mozilla.org`
-- `inv setup`
-
-If you're on windows, you need an extra step: run `inv manage createsuperuser` to create an admin user.
+- `inv new-env`
```
[the remark this answers is not in the document]
Comment (cadecairos) [reply README.md:41]: yeah I think this makes sense

[the remark this answers is not in the document]
Comment (cadecairos) [reply README.md:41]: we'll keep it aliased for historical reasons.

[the remark this answers is not in the document]
Comment (Pomax) [reply README.md:41]: +1 to alias.

**Comment by Pomax** [comment]
looks like it was some kind of `npm` hiccup, it works now, although it does take a very long time to run through `new-env` so I'm doing some timing tests.

I did notice that running `new-env` twice results in this error:

```
C:\Users\Mike\Documents\Git\mozilla\foundation.mozilla.org>inv new-env
* Setting default environment variables
* Stripping quotes and making sure your DATABASE_URL and ALLOWED_HOSTS are properly setup
* Setting default environment variables
* Stripping quotes and making sure your DATABASE_URL and ALLOWED_HOSTS are properly setup
* Stopping project's containers and delete volumes if necessary
cd C:\Users\Mike\Documents\Git\mozilla\foundation.mozilla.org && docker-compose down --volumes
Stopping foundationmozillaorg_postgres_1 ...
Removing foundationmozillaorg_postgres_1               ... 
Removing foundationmozillaorg_backend_run_759c440b6b90 ...
Removing foundationmozillaorg_watch-static-files_1     ...
Removing foundationmozillaorg_postgres_1               ...
Removing volume foundationmozillaorg_postgres_data ... done
* Building Docker images
Removing network foundationmozillaorg_default1     ... done
Removing volume foundationmozillaorg_postgres_data
* Building Docker images
cd C:\Users\Mike\Documents\Git\mozilla\foundation.mozilla.org && docker-compose build
postgres uses an image, skipping
Building watch-static-files
Step 1/2 : FROM node:12.16.2-stretch-slim
 ---> bb86086cce7c
Step 2/2 : WORKDIR /app/
 ---> Using cache
 ---> 045fd2e48b45
Successfully built 045fd2e48b45
Successfully tagged foundationmozillaorg_watch-static-files:latest
Building backend
Step 1/5 : FROM python:3.7.7-slim
 ---> d86f161fda14
Step 2/5 : RUN apt-get update && apt-get install -y       gettext     --no-install-recommends && rm -rf /var/lib/apt/lists/*
 ---> Using cache
 ---> 8dfb9046bcb6
Step 3/5 : ENV PYTHONUNBUFFERED 1
 ---> Using cache
 ---> a4310123390a
Step 4/5 : ENV PYTHONDONTWRITEBYTECODE 1
 ---> Using cache
 ---> 6ca36174f12c
Step 5/5 : WORKDIR /app/
 ---> Using cache
 ---> 53fbd30b30ee
Successfully built 53fbd30b30ee
Successfully tagged foundationmozillaorg_backend:latest
* Install Node dependencies
cd C:\Users\Mike\Documents\Git\mozilla\foundation.mozilla.org && docker-compose run --rm watch-static-files npm ins
tall
Creating network "foundationmozillaorg_default" with the default driver
Creating volume "foundationmozillaorg_postgres_data" with default driver
npm ERR! code EEXIST
npm ERR! syscall mkdir
npm ERR! path /app/node_modules
npm ERR! errno -17
npm ERR! EEXIST: file already exists, mkdir '/app/node_modules'
npm ERR! File exists: /app/node_modules
npm ERR! Remove the existing file and try again, or run npm
npm ERR! with --force to overwrite files recklessly.
```

**Comment by Pomax** [comment]
Retrying, I'm now getting this error:
```
[...]
Step 3/5 : ENV PYTHONUNBUFFERED 1
 ---> Running in 61e18f5e0d39
Removing intermediate container 61e18f5e0d39
 ---> b76e7a2bad54
Step 4/5 : ENV PYTHONDONTWRITEBYTECODE 1
 ---> Running in cce16199437a
Removing intermediate container cce16199437a
 ---> 9a29d90a2e93
Step 5/5 : WORKDIR /app/
 ---> Running in 3f316bd7d7f8
Removing intermediate container 3f316bd7d7f8
 ---> d9c5ba6a17f7
Successfully built d9c5ba6a17f7
Successfully tagged foundationmozillaorg_backend:latest
* Install Node dependencies
cd C:\Users\Mike\Documents\Git\mozilla\foundation.mozilla.org && docker-compose run --rm watch-static-files npm ins
tall
Creating network "foundationmozillaorg_default" with the default driver
Creating volume "foundationmozillaorg_postgres_data" with default driver
npm ERR! code ENOENT
npm ERR! syscall rename
npm ERR! path /app/node_modules/.staging/stylelint-b158a89c
npm ERR! dest /app/node_modules/stylelint
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, rename '/app/node_modules/.staging/stylelint-b158a89c' -> '/app/node_modules/stylelint'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent
```
So it feels like something is not right here, almost as if something's not waiting for `npm install` to finish before moving on to the next step

**Comment by Pomax** [comment]
Timing result for the new-env task:
- current version: 4.5 minutes
- this PR: 15 minutes

Timing result for then immediately running catchup:
- current version: 40 seconds
- this PR: 3 minutes

These differences are pretty drastic, do we know why piptools takes so much longer?

**Comment by cadecairos** [comment]
Can you share more about the environment you're running this on? 

Windows version, Docker version, command shell, Python & invoke versions

**Comment by Pomax** [comment]
- Windows 10 x64 pro, using `cmd` (through VS Code's terminal view)
- local python 3.7.7
- local invoke 1.4.1
- Docker for windows 2.2.0.5

**Comment by cadecairos** [comment]
> * Windows 10 x64 pro, using `cmd` (through VS Code's terminal view)
> 
> * local python 3.7.7
> 
> * local invoke 1.4.1
> 
> * Docker for windows 2.2.0.5

I'm running Python 3.8.1 and Docker 2.3.0.0 (Edge - so I can run WSL2 backed containers)

**Comment by cadecairos** [comment]
Attempting to install using cmd.

**Comment by cadecairos** [comment]
@Pomax I think this is a problem with the speed of writing to folders in the docker container that are mounted from Windows (node_modules and dockerpythonvenv

I could recommend using WSL2 and Docker's WSL2 integration - speeds are near native - but that's not fair because they're not release software yet (I think). It might be better performance wise to keep commands / documentation to run python and node things natively - while still running services in docker?

**Comment by Pomax** [comment]
It might make sense to shelve this work for the moment, so we can figure out what other options might be (not necessarily different from piptools, but from where "the files" live). Keeping the environment in the Docker guest would be fine, as long as `invoke` still works on the host - the only thing that needs to run outside of Docker right now is node, for accessibility/cypress testing, so if it's possible to "desync" that one directly, where Docker keeps the `node_modules` dir out of the share so that it can do what it needs, and the user on the host can do their own `npm install` and npm task running, that would be perfectly fine (as Cypress testing is something one already has to manually do).

**Comment by patjouk** [comment]
@cadecairos To test the `inv new-db`, you will need to delete your DB volume:
- `docker volume ls`  # Find the postres_data volume for the foundation site
- `docker volume rm [VOLUME NAME]`  # Replace VOLUME NAME by the one you got from the ls

Why? Because I'm removing the password for the db. I can't do `docker-compose down --volume` in the `new-db` task anymore since it's also gonna destroy the volume containing the python venv. I need to drop the db, which in the previous setup, would ask me for a password, which would have break on windows. I could have set the `PGPASSWORD` env just before calling the command, but again, it's different on windows and linux. I decided to save us some trouble by overriding the need for a password. We don't use any of this on prod, so we're good. If you don't delete your volume, it still gonna ask for a password.

@Pomax Requesting another review of this. Tell me if that fixes your performance issues.

**Comment by Pomax** [comment]
since we fully control the docker images, though, can't we simply use the default postgres:postgres username and password and supply those values in our dropdb command? There's nothing to "expose" in this case.

**Comment by cadecairos** [comment]
> delete your DB volume

I ran `docker system prune --all` to make sure I didn't have any collisions with containers I created in my project directory in my WSL directory. That should have removed the database volume.

Also, I wasn't prompted for password, I was actually just asked to confirm running the localization migration commands (iirc)

**Comment by patjouk** [comment]
We I ran the dropdb command when we had setup a password, I was asked to give one. Everytime I tried to connect with psql, I had to provide a password. Can you try another branch to see if it's the same for you?

**Comment by patjouk** [comment]
While I look again at the password situation, @pomax can you run `inv new-env` again and tell if it solves your performance issues?

**Comment by Pomax** [comment]
In order for it to run I had to add `POSTGRES_PASSWORD=mozilla` back into the docker-compose.yml, since otherwise the new-env task breaks on `re.search("POSTGRES_PASSWORD=(.*)", docker_compose).group(1) or password`, but with that done: _definitely_ a lot faster than when I tried it before, this is about as fast as current master! \o/

**Comment by patjouk** [comment]
If you try to initiate a postgres DB without a password, you will get this message:
```
postgres_1            | Error: Database is uninitialized and superuser password is not specified.
postgres_1            |        You must specify POSTGRES_PASSWORD to a non-empty value for the
postgres_1            |        superuser. For example, "-e POSTGRES_PASSWORD=password" on "docker run".
postgres_1            |
postgres_1            |        You may also use "POSTGRES_HOST_AUTH_METHOD=trust" to allow all
postgres_1            |        connections without a password. This is *not* recommended.
postgres_1            |
postgres_1            |        See PostgreSQL documentation about "trust":
postgres_1            |        https://www.postgresql.org/docs/current/auth-trust.html
```

It's a change that was made on postgres docker images to make them more secure ([more context here](http://support.divio.com/en/articles/3719228-database-is-uninitialized-and-superuser-password-is-not-specified)). Reverting to defaults won't change anything since a password will still be required.

I removing the code that's looking for a password in our `docker-compose.yml`  and keeping the changes I made in the `docker-compose.yml`.

**Comment by patjouk** [comment]
I've updated dependencies and fix some issues detected by the new version of flake8.
Ready for final review.

**Comment by Pomax** [comment]
This looks like it's working great, and importing staging/prod copies still works, too. The local requirements for Cypress also still work, so: R+!

**Pull request merged**